ui/panels-plugin/src/plugins/line-chart/LineChart.tsx

```diff
@@ -11,23 +11,124 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {
-  AnyChartQueryDefinition,
-  JsonObject,
-  PanelProps,
-} from '@perses-ui/core';
+import * as echarts from 'echarts/core';
+import type { EChartsOption } from 'echarts';
+import { LineChart as EChartsLineChart } from 'echarts/charts';
+import { GridComponent, DatasetComponent } from 'echarts/components';
+import { CanvasRenderer } from 'echarts/renderers';
+import { useMemo, useState, useLayoutEffect, useRef } from 'react';
+import { Box } from '@mui/material';
+import { useRunningGraphQueries } from './GraphQueryRunner';
 
-export const LineChartKind = 'LineChart' as const;
+echarts.use([
+  EChartsLineChart,
+  GridComponent,
+  DatasetComponent,
+  CanvasRenderer,
+]);
 
-type LineChartKind = typeof LineChartKind;
+export interface LineChartProps {
+  width: number;
+  height: number;
+}
 
-export type LineChartProps = PanelProps<LineChartKind, LineChartOptions>;
+/**
+ * Draws a LineChart with Apache ECharts for the current running time series.
+ */
+function LineChart(props: LineChartProps) {
+  const { width, height } = props;
+  const queries = useRunningGraphQueries();
 
-interface LineChartOptions extends JsonObject {
-  query: AnyChartQueryDefinition;
-  show_legend?: boolean;
-}
+  // Calculate the LineChart options based on the query results
+  const option: EChartsOption = useMemo(() => {
+    const dataset: EChartsOption['dataset'] = [];
+    const series: EChartsOption['series'] = [];
+
+    for (const query of queries) {
+      // Skip queries that are still loading and don't have data
+      if (query.loading || query.data === undefined) continue;
+
+      // For every series that comes back from a query, add a Dataset and a Series
+      // to the chart
+      let id = 0;
```

Comment:
Should the `id` be incremented per query, not per series? I'm getting a `Duplicated option on id "0"` error when trying to run multiple queries currently.

Reply:
Although I guess that error would indicate that it really needs to be different for every series, even across queries?